frontend/src/metabase/modes/components/drill/ColumnFilterDrill.jsx

```diff
@@ -22,15 +22,8 @@ export default function ColumnFilterDrill({ question, clicked }) {
     return [];
   }
 
-  const { column } = clicked;
-
-  // if our field ref doesn't match our column id, we have a remapped column
-  // and need to use the field id instead of the ref
-  const fieldRef =
-    column.id !== undefined && column.id !== column.field_ref[1]
-      ? ["field", column.id, null]
```

Comment:
`column.id !== undefined && column.id !== column.field_ref[1]` breaks the scenario as described by #22715 because the `column` in this scenario has an integer `id` that references the Field we've mapped to in the Model metadata sidebar, but the `field_ref` is in the form `["field", "ALIAS_CREATED_AT", {...}]`.

UI components should no be so introspective of columns/fields anyways, so I think it is preferable to rely on a simpler `dimension.mbql()` to properly generate a `field_ref`.

Reply:
@iethree -- I don't want to break the fix from https://github.com/metabase/metabase/pull/24622. Is there anything else I should test besides what's in the "Testing Steps"? Seems like that still works.

Reply:
👍  this is definitely the better solution. I only did it that way because I didn't have access to `dimension.mbql()`